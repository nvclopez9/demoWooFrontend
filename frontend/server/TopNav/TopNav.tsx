import { Logo } from '@woographql/server/Logo';
import { NavLink } from '@woographql/ui/NavLink';
import { UserNav } from '@woographql/client/UserNav';

export interface NavItem {
  label: string;
  href: string;
  cta?: boolean;
}

export interface TopNavProps {
  menu: NavItem[];
}

export function TopNav({ menu }: TopNavProps) {
  return (
    <nav className="w-full bg-white min-h-24 py-4 px-4">
      <ul className="max-w-screen-lg m-auto w-full flex flex-row gap-x-4 justify-end items-center">
        <Logo className="mr-auto" />
        {menu.map((item, i) => (
          <li key={i} className="group">
            <NavLink href={item.href}>
              {item.label}
            </NavLink>
          </li>
        ))}
        <UserNav />
      </ul>
    </nav>
  )
}