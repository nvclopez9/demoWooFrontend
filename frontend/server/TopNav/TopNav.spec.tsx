// TopNav.test.tsx
import { render, screen, mockedMenu } from '@woographql/testing';
import { TopNav } from './TopNav';

describe('TopNav component', () => {
  process.env.FRONTEND_URL = 'http://localhost';
  it('renders correctly', () => {
    render(<TopNav menu={mockedMenu} />);

    // Check that the logo is rendered
    expect(screen.getByRole('img')).toBeInTheDocument();

    // Check that the menu items are rendered
    mockedMenu.forEach((item) => {
      expect(screen.getByText(item.label)).toBeInTheDocument();
    });

    // Check that the user navigation is rendered
    expect(screen.getByText('Cart')).toBeInTheDocument();
    expect(screen.getByText('Checkout')).toBeInTheDocument();
    expect(screen.getByText('Login')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<TopNav menu={mockedMenu} />);
    expect(baseElement).toMatchSnapshot();
  });
});
