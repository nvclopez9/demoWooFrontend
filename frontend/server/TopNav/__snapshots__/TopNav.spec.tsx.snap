// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav component should match snapshot 1`] = `
<body>
  <div>
    <nav
      class="w-full bg-white min-h-24 py-4 px-4"
    >
      <ul
        class="max-w-screen-lg m-auto w-full flex flex-row gap-x-4 justify-end items-center"
      >
        <a
          class="mr-auto inline-flex items-center"
          href="http://localhost"
        >
          <svg
            class="w-16 h-16"
            role="img"
            viewBox="0 0 177.27899 177.17758"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <lineargradient
                id="linearGradient4987"
              >
                <stop
                  offset="0"
                  style="stop-color: rgb(69, 123, 157); stop-opacity: 1;"
                />
                <stop
                  offset="1"
                  style="stop-color: rgb(69, 123, 157); stop-opacity: 0;"
                />
              </lineargradient>
              <lineargradient
                id="linearGradient4540"
              >
                <stop
                  offset="0"
                  style="stop-color: rgb(241, 250, 238); stop-opacity: 1;"
                />
                <stop
                  offset="1"
                  style="stop-color: rgb(69, 123, 157); stop-opacity: 1;"
                />
              </lineargradient>
            </defs>
            <g
              transform="translate(-24.196996,-62.039592)"
            >
              <path
                class="ghost-body"
                d="m 92.423219,64.549439 c -36.73501,10.57719 -58.073058,51.510361 -47.085986,86.628681 10.987072,35.11832 38.813875,61.19825 55.491517,87.4188 l 24.57751,-60.42174 52.85073,38.1278 c 0,0 7.81634,-71.22793 0.61813,-103.57447 -7.16614,-32.202429 -49.71686,-58.756107 -86.4519,-48.17903 z m -26.40159,53.160201 c 14.03161,-4.03995 28.681541,4.0597 32.721818,18.09121 4.040283,14.03172 -4.059352,28.68198 -18.09106,32.72232 -14.031917,4.04034 -28.682348,-4.05958 -32.72246,-18.09156 -4.04001,-14.0318 4.059909,-28.68191 18.091702,-32.72197 z M 143.44987,95.415697 c 14.03172,-4.040284 28.68198,4.059358 32.72231,18.091063 4.04034,14.03192 -4.05957,28.68235 -18.09155,32.72246 -14.03181,4.04001 -28.68192,-4.05991 -32.72197,-18.0917 -4.03995,-14.03161 4.05969,-28.681545 18.09121,-32.721823 z"
                style="fill: #e63946; fill-opacity: 1;"
              />
              <g
                class="ghost-eyes"
                style="display: inline; fill: #457b9d; fill-opacity: 1; stroke: url(#linearGradient4987); stroke-opacity: 1;"
              >
                <path
                  class="ghost-left-eye"
                  d="m 66.021629,117.70964 c 14.03161,-4.03995 28.681541,4.0597 32.721818,18.09121 4.040283,14.03172 -4.059352,28.68198 -18.09106,32.72232 -14.031917,4.04034 -28.682348,-4.05958 -32.72246,-18.09156 -4.04001,-14.0318 4.059909,-28.68191 18.091702,-32.72197 z"
                />
                <path
                  class="ghost-right-eye"
                  d="m 143.44987,95.415697 c 14.03172,-4.040284 28.68198,4.059358 32.72231,18.091063 4.04034,14.03192 -4.05957,28.68235 -18.09155,32.72246 -14.03181,4.04001 -28.68192,-4.05991 -32.72197,-18.0917 -4.03995,-14.03161 4.05969,-28.681545 18.09121,-32.721823 z"
                />
              </g>
              <g
                class="ghost-inner-eyes"
                style="display: inline; fill: url(#linearGradient4540); fill-opacity: 1; stroke: none; stroke-opacity: 1;"
                transform="translate(2.6458334,7.4083337)"
              >
                <path
                  class="ghost-inner-right-eye"
                  d="m 146.24401,105.1196 c 8.67241,-2.49712 17.7271,2.50892 20.22426,11.18131 2.49716,8.67253 -2.50905,17.72733 -11.18161,20.22435 -8.67246,2.49696 -17.72707,-2.50925 -20.22404,-11.1817 -2.49692,-8.67234 2.50911,-17.72684 11.18139,-20.22396 z"
                />
                <path
                  class="ghost-inner-left-eye"
                  d="m 68.815627,127.41373 c 8.672267,-2.4969 17.72669,2.5091 20.223792,11.18131 2.497105,8.67233 -2.508892,17.72696 -11.18122,20.2241 -8.672457,2.49714 -17.727188,-2.50903 -20.224189,-11.18153 -2.496937,-8.67238 2.509236,-17.72691 11.181617,-20.22388 z"
                />
              </g>
            </g>
          </svg>
          <h1
            class="m-0 font-serif text-3xl font-bold text-peppermint"
          />
        </a>
        <li
          class="group"
        >
          <a
            class="transition-colors group-hover:text-blue-400"
            href="/clothing"
          >
            Clothing
          </a>
        </li>
        <li
          class="group"
        >
          <a
            class="transition-colors group-hover:text-blue-400"
            href="/accessories"
          >
            Accessories
          </a>
        </li>
        <li
          class="group"
        >
          <a
            class="transition-colors group-hover:text-blue-400"
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li
          class="group"
        >
          <button
            class="justify-center rounded-md focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 h-9 flex flex-row gap-x-2 items-center p-0 hover:no-underline text-base font-normal transition-colors group-hover:text-blue-400"
            data-testid="cart-button"
          >
            0
            <i
              aria-hidden="true"
              class="fa-solid fa-basket-shopping"
            />
            <span
              class="transition-all origin-left w-0 scale-x-0 group-hover:scale-x-100 group-hover:w-8"
            >
              Cart
            </span>
          </button>
        </li>
        <li
          class="group w-auto"
        >
          <button
            class="justify-center rounded-md focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 h-9 flex flex-row gap-x-2 items-center p-0 hover:no-underline text-base font-normal transition-colors group-hover:text-blue-400"
            data-testid="checkout-button"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-cash-register"
            />
            <span
              class="transition-all origin-left w-0 scale-x-0 group-hover:scale-x-100 group-hover:w-16"
            >
              Checkout
            </span>
          </button>
        </li>
        <li
          class="group"
        >
          <a
            class="flex flex-row gap-x-2 items-center transition-colors group-hover:text-blue-400"
            href="/login"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-arrow-right-to-bracket"
            />
            <span
              class="transition-all origin-left w-0 scale-x-0 group-hover:scale-x-100 group-hover:w-8"
            >
              Login
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
